src/SprykerSdk/Zed/ComposerConstrainer/Business/Validator/StrictConstraintValidator.php

```diff
@@ -143,8 +143,14 @@ protected function setDefinedModuleInfo(
             return $usedModuleInfo;
         }
 
-        $version = (string)($definedConstraint ? $definedConstraint->getVersion() : $featureConstraint->getVersion());
-        $isDev = (bool)($definedConstraint ? $definedConstraint->getIsDev() : $featureConstraint->getIsDev());
+        $constraint = $definedConstraint;
+        if ($constraint === null) {
+            /** @var \Generated\Shared\Transfer\ComposerConstraintTransfer $constraint */
```

Comment:
This should not be necessary if the $featureConstraint cannot be null - if it was already checked for this.

Reply:
phpstan can't figure it out